ENIGMAsystem/SHELL/Graphics_Systems/General/GSmodel.cpp

```diff
@@ -212,11 +214,18 @@ void d3d_model_draw(int id, gs_scalar x, gs_scalar y, gs_scalar z, int texId) {
 
 void d3d_model_primitive_begin(int id, int kind, int format) {
   enigma::Model* model = enigma::models[id];
+  #ifdef DEBUG_MODE
+  if (model->current_primitive) {
+    show_error("d3d_model_primitive_begin called before d3d_model_primitive_end " \
+               "on the previous primitive for model: " + std::to_string(id), false);
+    return;
+  }
+  #endif
   if (!model->vertex_started) {
     model->vertex_started = true;
     vertex_begin(model->vertex_buffer);
   }
-
+  model->vertex_colored = true;
   model->current_primitive = new enigma::Primitive(
```

Comment:
Unless you're handing this memory off to someone, lose these `new` calls. You think you're saving memory doing this, but really you're just causing the game to have to reallocate the next time they draw this many vertices.

Reply:
Don't conflate issues please, I'll look at this in a separate PR. I thought that may be a great case to optimize the create time back up again though. It is not handed off, and is actually copied to the stack and then deleted in the primitive end before I emplace the stack copy into the primitives vector.